sympy/solvers/solvers.py

```diff
@@ -948,52 +1021,42 @@ def _sympified_list(w):
                 else Abs(arg))
             if was == fi:
                 break
-
         for e in fi.find(Abs):
             if e.has(*symbols):
                 raise NotImplementedError('solving %s when the argument '
                     'is not real or imaginary.' % e)
-
         # arg
         fi = fi.replace(arg, lambda a: arg(a).rewrite(atan2).rewrite(atan))
-
         # save changes
         f[i] = fi
-
     # see if re(s) or im(s) appear
-    freim = [fi for fi in f if fi.has(re, im)]
```

Comment:
Why was this changed?

Reply:
This was actually dropped in your PR. Probably because dropping this line doesn't change the output.

Reply:
I just noticed that there are too few tests on real/imaginary handling in `test_solver.py`. So, we need to incorporate more tests to see whether the behaviour remains same or not.

Reply:
I don't see it in the diff from the PR:
https://github.com/sympy/sympy/pull/17993/files

Reply:
For some reason, the line isn't there in the diff. But it is absent from the file. 
https://github.com/sympy/sympy/pull/17993/files#diff-d914c8b9dd231a57e757eb21513e3cff87b0aba6fbe3dc9ad128a3c16ad03e4bR1137

Reply:
Oh, I see! That's a big PR and I need to go through it and understand it before making further changes here.

Reply:
I don't think that you need to understand all of #18844. The main thing is that this PR should only bring forward changes that I made in #17993 or any changes that you have decided to make.

I'm not sure how you created the initial commit here but I would have done it using `git rebase`: that can "rebase" the diff on top of any changes made since. Instead it looks you've copied bits of the file from that PR but that has the potential to undo other changes. That means you need to check through the changes here carefully to see whether they are changes that should be here.

Reply:
Yes you are right, I copied bits of the file, since this was the first time I brought changes from another PR to mine.